eth/downloader/downloader.go

```diff
@@ -1380,6 +1390,13 @@ func (d *Downloader) processHeaders(origin uint64, td, ttd *big.Int, beaconMode
 							rollback = 1
 						}
 					}
+					if len(rejected) != 0 {
+						// Merge threshold reached, stop importing, but don't roll back
+						rollback = 0
+
+						log.Info("Legacy sync reached merge threshold", "number", rejected[0].Number, "hash", rejected[0].Hash(), "td", td, "ttd", ttd)
```

Comment:
The `number` reported here is the first rejected, so, 
` block-n, transition-block, invalid-block-1,...`, the `invalid-block-1` is what it will output. Seems not the right thing to output?

Reply:
Actually, that's exactly what I wanted to output. The block that's imported is not that relevant since we have import logs too as well as it's part of the chain so can be retrieved. What I'm interested to see is the details of the block that is rejected so I can look it up or investigate.

Reply:
But -- if all our peers are well-behaving, they won't send us PoS blocks over the network, will they? Just want to avoid the scenario where we wait for the fist non-pow header to abort our cycle and "officially enter beacon mode", but our peers aren't delivering it because it's PoS.